### PR TITLE
feat: first implementation of publish_module extrinsic

### PR DESCRIPTION
**Pull request by asmie**
First usage of `move-vm-backend` inside this pallet - VM is made inside the `publish_module` extrinsic, and then the module can be published.

Still to be done - creating MoveVM should be separated and the results (especially errors) should be more accurate if something fails.

Added test to check if it's working correctly.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -21,9 +21,16 @@ pub mod pallet {
         pallet_prelude::*,
     };
     use frame_system::pallet_prelude::*;
+    // We need now use MoveStorage which provides ModuleResolver and ResourceResolver.
+    // Once we deal with it in move_vm_backend and storage will not depened upon those
```
Comment (Rqnsom) [inline src/lib.rs:25]:
```suggestion
    // Once we deal with it in move_vm_backend and storage will not depend upon those
```

**Review by Rqnsom — APPROVED** [review]
LGTM

File: src/lib.rs [hunk 2]
```diff
@@ -92,13 +99,22 @@ pub mod pallet {
...
+            bytecode: Vec<u8>,
             _gas_limit: u64,
         ) -> DispatchResultWithPostInfo {
             // Allow only signed calls.
             let who = ensure_signed(origin)?;
 
-            // TODO: Publish module
+            let storage = MoveStorage::new(Self::move_vm_storage());
+
+            let vm = Mvm::new(storage).map_err(|_err| Error::<T>::PublishModuleFailed)?;
```
[the remark this answers is not in the document]
Comment (asmie) [reply src/lib.rs]: Done.

File: src/lib.rs (hunk 1 above)
Comment (asmie) [reply src/lib.rs]: Done.

**Review by Rqnsom — APPROVED** [review]

**Pull request merged**